mono_repo/lib/src/mono_config.dart

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library mono_repo.travis_config;
+
+import 'package:collection/collection.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'mono_config.g.dart';
+
+final monoFileName = '.mono_repo.yml';
+final travisFileName = '.travis.yml';
+final travisShPath = './tool/travis.sh';
+
+@JsonSerializable()
+class MonoConfig extends Object with _$MonoConfigSerializerMixin {
+  @override
+  final List<String> sdks;
+
+  @override
+  final List<String> stageNames;
+
+  @override
+  final List<TravisJob> jobs;
+
+  MonoConfig(this.sdks, this.stageNames, this.jobs);
+
+  factory MonoConfig.parse(String package, Map<String, dynamic> monoYaml) {
+    var unrecognizedKeys =
+        (monoYaml.keys.toSet()..removeAll(_validKeys)).toList()..sort();
+
+    if (unrecognizedKeys.isNotEmpty) {
+      throw new ArgumentError(
+          'Unrecognized keys $unrecognizedKeys in .mono_repo.yaml.'
+          'Only $_validKeys are allowed.');
+    }
+
+    var sdks = (monoYaml['dart'] as List<String>)?.toList();
+    if (sdks == null || sdks.isEmpty) {
+      throw new ArgumentError(
+          'At least one SDK version is required under "dart".');
```

Comment:
would it be reasonable to default to "dev"?

Reply:
We didn't support this previously, I think the main issue is its not clear whether the default should be "dev" or "stable"?

Reply:
Requiring it to be explicit is fine with me.